liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Misc.hs

```diff
@@ -1007,3 +1007,19 @@ prependGHCRealQual = varQual_RDR gHC_INTERNAL_REAL
 
 isFromGHCReal :: NamedThing a => a -> Bool
 isFromGHCReal x = Ghc.nameModule (Ghc.getName x) == gHC_INTERNAL_REAL
+
+-- Collect all variables (free and bound) in a CoreExpr. Couldn't find in in GHC libs so here it is
+collectAllVars :: CoreExpr -> S.HashSet Var
+collectAllVars = go S.empty
```

Comment:
The one from GHC did'nt yield enough variables, so I reimplemented it. And I thought it'd be simpler to just collect all variables since in any case we just use it as bag of variables to use to parse the logic expression. But I've changed it so at least we remove variables bound by a lambda expression